src/components/Modals/Send/hooks/useSendDetails/useSendDetails.tsx

```diff
@@ -152,130 +170,147 @@ export const useSendDetails = (): UseSendDetailsReturnType => {
         if (estimatedFees !== undefined) {
           setValue(SendFormFields.AmountFieldError, '')
           setValue(SendFormFields.EstimatedFees, estimatedFees)
         }
-      },
-      1000,
-      { leading: true, trailing: true },
-    )
-  }, [
-    accountId,
-    asset,
-    assetId,
-    cryptoHumanBalance,
-    estimateFormFees,
-    feeAsset.assetId,
-    feeAsset.symbol,
-    getValues,
-    nativeAssetBalance,
-    setValue,
-  ])
 
-  // Stop calls to debouncedSetEstimatedFormFees on unmount
-  useEffect(() => () => debouncedSetEstimatedFormFees.cancel(), [debouncedSetEstimatedFormFees])
+        return estimatedFees
+      } catch (e) {
+        throw new Error('common.insufficientFunds')
+      }
+    },
+    [
+      accountId,
+      asset,
+      assetId,
+      cryptoHumanBalance,
+      estimateFormFees,
+      feeAsset.assetId,
+      feeAsset.symbol,
+      nativeAssetBalance,
+      sendMax,
+      setValue,
+    ],
+  )
+
+  const queryKey = useDebounce(
+    () => ['setEstimatedFormFees', { amountCryptoPrecision }],
+    1000,
+  ) as unknown as [string, { amountCryptoPrecision: string }]
+
+  const {
+    isLoading: _isEstimatedFormFeesLoading,
+    data: estimatedFees,
+    error,
+  } = useQuery({
+    queryKey,
+    queryFn: setEstimatedFormFeesQueryFn,
+    enabled: true,
+    // a very arbitrary 15 seconds, which is enough to cache things in case the user is having fun with the input,
+    // but also safe to invalidate in case there's a new Tx changing their balance
+    staleTime: 15 * 1000,
+    // Consider failed queries as fresh, not stale, and don't do the default retry of 3 for them, as failures *are* expected here with insufficient funds
+    retry: false,
+  })
+
+  const isTransitioning = useMemo(
+    () => queryKey[1].amountCryptoPrecision !== amountCryptoPrecision,
+    [amountCryptoPrecision, queryKey],
+  )
+
+  // Since we are debouncing the query, the query fire is delayed by however long the debounce is
+  // This would lead to delayed loading states visually, which look odd and would make users able to continue with a wrong state
+  const isEstimatedFormFeesLoading = isTransitioning || _isEstimatedFormFeesLoading
+
+  useEffect(() => {
+    setValue(SendFormFields.AmountFieldError, error?.message ? error.message : '')
+  }, [error, setValue])
+
+  useEffect(() => {
+    if (!estimatedFees) return
+    // sendMax sets its own fees
+    if (sendMax) return
+
+    setValue(SendFormFields.EstimatedFees, estimatedFees)
+  }, [estimatedFees, sendMax, setValue])
 
   const handleNextClick = () => history.push(SendRoutes.Confirm)
 
   const handleSendMax = async () => {
+    // Since we fetch the query internally here with different args, we can't leverage the loading state from the useQuery() above and need a separate state
+    // field to track loading state
+    setValue(SendFormFields.SendMax, true)
+    setIsFormLoading(true)
```

Comment:
This pattern creates a race condition where an initial call can set `isFormLoading` to false while a subsequent call is still in flight. We should either move this to a react-query, or guard against concurrent calls.

Reply:
@woodenfurniture while I do conceptually agree, that would be the same as virtually any async function that sets loading to true at its start, and false at the end.

Most notably, this is no different than the current implementation calling `setLoading(true)` before starting the heavy duty and then `setLoading(false)` when done (not that the current implementation does things well, that's the whole purpose of this PR!).

re: react-query, see the comment above on l.274, we can't call `refetch` from the `useQuery()` because of different args (i.e max balance vs. whatever is currently input).

Either way, while this may have been a potential concern previously and we had a lot of general issues in the send form re: loading states and stale data, this is now a non-concern as we do have a staleTime of 15s for a given query, meaning clicking sendMax twice won't be an issue - data will be gotten from the cache immediately when double-clicking "Max" (and the same goes for the component-level `useQuery()`), meaning things do load immediately on second try.

See this Jam https://jam.dev/c/c5ab282a-d47a-4a7c-824f-0158c321f4de (note, the double fees estimation for UTXOs *is* part of the flow: first estimation in the internal query here for a max send, then component-level `useQuery` estimation for the fee-deducted amount).